src/Streamly/Internal/Prelude.hs

```diff
@@ -751,10 +751,8 @@ repeatMSerial = fromStreamS . S.repeatM
 -- @
 --
 -- @since 0.1.2
-iterate :: IsStream t => (a -> a) -> a -> t m a
-iterate step = K.fromStream . go
-    where
-    go s = K.cons s (go (step s))
+iterate :: (Monad m, IsStream t) => (a -> a) -> a -> t m a
```

Comment:
Instead of removing the old implementation of `iterate` and `iterateM` move it to the StreamK module. Here use `fromStreamS . S.iterate` so that appropriate implementation (StreamD or StreamK) can be picked based on the `USE_STREAMK_ONLY` cpp macro.